File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/KuduzProfilaksiIzlemFormViewModel.cs
//$C8FE8645
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Infrastructure.Helpers;

namespace Core.Controllers
{
    public partial class KuduzProfilaksiVeriSetiServiceController
    {
        partial void PreScript_KuduzProfilaksiIzlemForm(KuduzProfilaksiIzlemFormViewModel viewModel, KuduzProfilaksiVeriSeti kuduzProfilaksiVeriSeti, TTObjectContext objectContext)
        {
            Guid? selectedEpisodeActionID = viewModel.GetSelectedEpisodeActionID();
            if (selectedEpisodeActionID.HasValue)
            {
                EpisodeAction episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionID.Value);
                viewModel._KuduzProfilaksiVeriSeti.EpisodeAction = episodeAction;
                viewModel._KuduzProfilaksiVeriSeti.SubEpisode = episodeAction.SubEpisode;

            }
            ContextToViewModel(viewModel, objectContext);
        }

    }
}

namespace Core.Models
{
    public partial class KuduzProfilaksiIzlemFormViewModel: BaseViewModel, IENabizViewModel
    {
        public void AddENabizObjectViewModelToContext(TTObjectContext objectContext)
        {
            objectContext.AddObject(this._KuduzProfilaksiVeriSeti);

            if (this.KuduzProfilaksiUygProfilakGridList!= null)
            {
                foreach (var item in this.KuduzProfilaksiUygProfilakGridList)
                {
                    var kuduzProfilaksiUygProfilak = (KuduzProfilaksiUygProfilak)objectContext.AddObject(item);
                    kuduzProfilaksiUygProfilak.KuduzProfilaksiVeriSeti = this._KuduzProfilaksiVeriSeti;
                }
            }


            var kuduzProfilaksiIzlemVeriSeti = (KuduzProfilaksiVeriSeti)objectContext.GetLoadedObject(this._KuduzProfilaksiVeriSeti.ObjectID);


            new SendToENabiz(objectContext, kuduzProfilaksiIzlemVeriSeti.SubEpisode, kuduzProfilaksiIzlemVeriSeti.ObjectID, kuduzProfilaksiIzlemVeriSeti.ObjectDef.Name, "236", Common.RecTime());
        }
    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/KuduzProfilaksiIzlemFormViewModel.ts
//$9504D910
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { KuduzProfilaksiVeriSeti } from "NebulaClient/Model/AtlasClientModel";
import { KuduzProfilaksiUygProfilak } from "NebulaClient/Model/AtlasClientModel";
import { SKRSImmunglobulinTuru } from "NebulaClient/Model/AtlasClientModel";
import { SKRSKuduzProfilaksisiTamamlanmaDurumu } from "NebulaClient/Model/AtlasClientModel";
import { SKRSUygulananKuduzProfilaksisi } from "NebulaClient/Model/AtlasClientModel";
import { Type } from 'NebulaClient/ClassTransformer';

export class KuduzProfilaksiIzlemFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.KuduzProfilaksiIzlemFormViewModel, Core';

    @Type(() => KuduzProfilaksiVeriSeti)
    public _KuduzProfilaksiVeriSeti: KuduzProfilaksiVeriSeti = new KuduzProfilaksiVeriSeti();
    @Type(() => KuduzProfilaksiUygProfilak)
    public KuduzProfilaksiUygProfilakGridList: Array<KuduzProfilaksiUygProfilak> = new Array<KuduzProfilaksiUygProfilak>();
    @Type(() => SKRSImmunglobulinTuru)
    public SKRSImmunglobulinTurus: Array<SKRSImmunglobulinTuru> = new Array<SKRSImmunglobulinTuru>();
    @Type(() => SKRSKuduzProfilaksisiTamamlanmaDurumu)
    public SKRSKuduzProfilaksisiTamamlanmaDurumus: Array<SKRSKuduzProfilaksisiTamamlanmaDurumu> = new Array<SKRSKuduzProfilaksisiTamamlanmaDurumu>();
    @Type(() => SKRSUygulananKuduzProfilaksisi)
    public SKRSUygulananKuduzProfilaksisis: Array<SKRSUygulananKuduzProfilaksisi> = new Array<SKRSUygulananKuduzProfilaksisi>();

   
}
